api/src/main/java/org/apache/iceberg/Transaction.java

```diff
@@ -137,6 +137,14 @@ default AppendFiles newFastAppend() {
    */
   DeleteFiles newDelete();
 
+  /**
+   * Create a new {@link UpdateTableStatistics update table statistics API} to add or remove statistics
+   * files in this table.
+   *
+   * @return a new {@link UpdateTableStatistics}
+   */
+  UpdateTableStatistics newUpdateTableStatistics();
```

Comment:
I think this should be called `updateTableStatistics`. No need for "new" because "update" is a verb.

Reply:
I guess i named this after newReplacePartitions or newOverwrite ("replace" and "overwrite" are also verbs).
Sure, will remove "new" here.

> Can you add the API to `Table` and `Transaction` in a separate PR?

Sure.
Should come together with or separate from BaseTransaction changes?
Should come together with or separate from BaseSnapshot changes?
Should come together with or separate from remove_orphan_files changes?

Reply:
Extracted https://github.com/apache/iceberg/pull/5021. Let me know if I got this right.

Reply:
I think we prefer not to add the implementation changes. Just the API addition with a default that throws an UnsupportedOperationException.